cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mysql/MySQLBackendImpl.java

```diff
@@ -329,135 +361,161 @@ public void expirateRecordsCache(long expirationTime) throws CygnusRuntimeError,
+
+        private final HashMap<String, DataSource> connections;
         private final String mysqlHost;
         private final String mysqlPort;
         private final String mysqlUsername;
         private final String mysqlPassword;
-        
+        private final int maxPoolSize;
+
         /**
          * Constructor.
+         * 
          * @param mysqlHost
          * @param mysqlPort
          * @param mysqlUsername
          * @param mysqlPassword
          */
-        public MySQLDriver(String mysqlHost, String mysqlPort, String mysqlUsername, String mysqlPassword) {
+        public MySQLDriver(String mysqlHost, String mysqlPort, String mysqlUsername, String mysqlPassword, int maxPoolSize) {
             connections = new HashMap<>();
             this.mysqlHost = mysqlHost;
             this.mysqlPort = mysqlPort;
             this.mysqlUsername = mysqlUsername;
             this.mysqlPassword = mysqlPassword;
+            this.maxPoolSize = maxPoolSize;
         } // MySQLDriver
-        
+
         /**
          * Gets a connection to the MySQL server.
+         * 
          * @param dbName
          * @return
+         * @throws CygnusRuntimeError
          * @throws CygnusPersistenceError
          */
         public Connection getConnection(String dbName) throws CygnusRuntimeError, CygnusPersistenceError {
             try {
-                // FIXME: the number of cached connections should be limited to a certain number; with such a limit
-                //        number, if a new connection is needed, the oldest one is closed
-                Connection con = connections.get(dbName);
+                // FIXME: the number of cached connections should be limited to
+                // a certain number; with such a limit
+                // number, if a new connection is needed, the oldest one is closed
+                Connection connection = null;
+
+                if (connections.containsKey(dbName)) {
+                    connection = connections.get(dbName).getConnection();
+                    LOGGER.debug("Recovered database connection from cache (" + dbName + ")");
+                }
 
-                if (con == null || !con.isValid(0)) {
-                    if (con != null) {
+                if (connection == null || !connection.isValid(0)) {
+                    if (connection != null) {
                         LOGGER.debug("Closing invalid mysql connection for db " + dbName);
-                        con.close();
+                        connection.close();
                     } // if
 
-                    con = createConnection(dbName);
-                    connections.put(dbName, con);
-                    LOGGER.debug("Number of cached mysql connections: " + connections.size());
+                    DataSource datasource = createConnectionPool(dbName);
+                    connections.put(dbName, datasource);
+                    connection = datasource.getConnection();
```

Comment:
Al principio o al final de este metodo pondria una traza con el tamaño de connections

Reply:
We aggre with this, modified in 102e8f1

https://github.com/telefonicaid/fiware-cygnus/blob/102e8f1940b44449fa69fa3786c767c4e0196f7a/cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/mysql/MySQLBackendImpl.java#L470-L477